### PR TITLE
WIP: Fix Format-Table/Format-Wide padding when output contains CJK characters

### PR DESCRIPTION
**Pull request by kwkam**
## PR Summary

Compute display width for the string in `LengthInBufferCells`, otherwise `Format-Table` and `Format-Wide` will pad unnecessary space for CJK character.

Add UTF-8 to checking list since it contains CJK characters.

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] `NA` User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] `NA` Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [x] `NA` [Add `[feature]` if the change is significant or affectes feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

**Comment by iSazonov** [comment]
@kwkam  Thanks for your contribution! Please follow the requirements of PR Checklist.

**Comment by lzybkr** [comment]
@kwkam - This might be fixed in the [PSReadLine repo](https://github.com/lzybkr/PSReadLine). You can install v2 from the [gallery](https://www.powershellgallery.com/packages/PSReadline/2.0.0-beta1).

I think the version in this repo will be removed in favor of pulling my version from the gallery, so if you still see issues, I suggest you submit a PR there instead.

The change to `ConsoleControl.cs` does make sense in this repo though, so we should merge that for sure.

**Comment by kwkam** [comment]
@lzybkr - Sorry I didn't know PSReadLine is an external module and yes the issue is fixed in upstream.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs [hunk 1]
```diff
@@ -2781,8 +2781,17 @@ internal static int LengthInBufferCells(string str, int offset, bool checkEscape
                     escapeSequenceAdjustment += ControlSequenceLength(str, ref i);
                 }
             }
+#if !UNIX
```
Comment (lzybkr) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs]: Isn't this a problem on Linux/Mac as well?

Comment (kwkam) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs]: I suppose but I cannot test.
Is there any reason why the `LengthInBufferCells` function is disabled on unix?

Comment (lzybkr) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs]: I don't know for sure, but I would guess the original code depended on Windows apis and when porting to non-Windows, they punted on porting that code until a later time.

Comment (kwkam) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs]: Exlcuding `LengthInBufferCells` from that block since it seems to be harmless.

**Pull request closed without merging**